src/GLuaFixer/LintMessage.hs

```diff
@@ -2,17 +2,39 @@ module GLuaFixer.LintMessage where
 
 import GLua.AG.Token
 import GLua.AG.PrettyPrint
+import Data.Char
 import Data.List (sortOn)
+import Text.ParserCombinators.UU.BasicInstances hiding (msgs)
 
 -- | Represents lint messages
 data LintMessage
   = LintError { err_rg :: Region, err_msg :: String, err_file :: String }
-  | LintWarning {warn_rg :: Region, warn_msg :: String, warn_file :: String }
+  | LintWarning { warn_rg :: Region, warn_msg :: String, warn_file :: String }
   deriving ( Eq )
 
 instance Show LintMessage where
-    show (LintError rg msg file) = showString file . showString ": [Error] " . showString (renderRegion rg) . showString ": " . showString msg $ ""
-    show (LintWarning rg msg file) = showString file . showString ": [Warning] " . showString (renderRegion rg) . showString ": " . showString msg $ ""
+    show (LintError rg msg file) = formatLintMessageDefault "Error" rg msg file
+    show (LintWarning rg msg file) = formatLintMessageDefault "Warning" rg msg file
+
+getLintMessageData :: LintMessage -> (Region, String, String, String)
+getLintMessageData (LintError rg msg file) = (rg, msg, file, "Error")
+getLintMessageData (LintWarning rg msg file) = (rg, msg, file, "Warning")
+
+formatLintMessage :: LintMessage -> String -> String
+formatLintMessage lintMsg format = do
+  let (rg, msg, file, level) = getLintMessageData lintMsg
```

Comment:
Instead of `getLintMessageData` you can do a case expression on `lintmsg`. It would look something like this:

```haskell
case lintMsg of
  LintError rg msg file -> ...
  LintWarning rg msg file -> ...
```

I think that case expression would fit best in `formatLintMessageGithub` and `FormatLintMessageDefault`